### DISCUSSION TITLE
"CodedEvent_SNOMED" error but codelist hasn't changed

### DISCUSSION THREAD

**Discussion by StevenMaude** [opensafely/documentation#675] [Q&A] [answered]
@ciaranmci [asked this question](https://github.com/opensafely/documentation/issues/673):
> Today I re-ran a script on the job server ([job ID: x5tvqrb4lpuiapa](https://jobs.opensafely.org/university-of-leeds/cancer-surgery-service-evaluation/all_cause_mortality_with_stratification/6577/x5tvqrb4lpuiapam/)) and, for the first time, my `generate_study_population` script failed citing an error code about "CodedEvent_SNOMED".
> I assume this refers to the `CodedEvent` table in the cohortextractor (but I'm only basing this assumption on [the only reference to "CodedEvent" that I could find](https://github.com/opensafely/documentation/issues/114)).
>
> My scripts run perfectly well on my local OpenSAFELY docker container, and my codelist hasn't changed since my most-recent successful run ([job ID: akakanz4oqjuokq4](https://jobs.opensafely.org/university-of-leeds/cancer-surgery-service-evaluation/all_cause_mortality_with_stratification/6514/akakanz4oqjuokq4/)).
>
>
> **Questions:**
>
>1. Can you direct me to the CodedEvent table in the hope that I might be able to cross check my codelist for possibly-missing code?
> 2. Do you have any ideas about what might suddenly have started causing this?

**Comment by ciaranmci** [comment]
Re-ran this morning and their was no problem ([job ID: cvhlmluih5oomwpu](https://jobs.opensafely.org/university-of-leeds/cancer-surgery-service-evaluation/all_cause_mortality_with_stratification/6608/cvhlmluih5oomwpu/)). Go figure!

Sorry to bother you with the move from issue to discussion, @StevenMaude.

**Reply by StevenMaude** [reply]
No problem @ciaranmci; glad you resolved it :smiley:

**Comment by evansd** [comment, marked as answer]
Hi @ciaranmci, sorry about this. Just to explain what happened here, there have been some issues recently doing the weekly data import as the number of active users of the platform has increased. As a temporary measure, TPP are dropping the `CodedEvent_SNOMED` table, rebuilding it, and then bringing it back online. This generally happens in the early hours of Monday morning and any queries which use SNOMED codelists will fail during this time. This obviously isn't ideal but it was better than the whole database locking up and there are plans for a better longer term solution.

We also need to provide a more helpful error messages to users so if someone else gets bitten by this they have some chance of understanding what's going on.

Again, sorry about this. But just to reassure you that it wasn't anything you did wrong and it shouldn't happen again as long as it isn't Monday!
